src/Microsoft.Health.Fhir.SqlServer/Features/Storage/SqlServerFhirDataStore.cs

```diff
@@ -668,9 +668,10 @@ public void Build(ICapabilityStatementBuilder builder)
         {
             EnsureArg.IsNotNull(builder, nameof(builder));
 
-            builder.PopulateDefaultResourceInteractions()
-                .SyncSearchParameters()
-                .AddGlobalSearchParameters()
+            var task = Task.Run(() => builder.PopulateDefaultResourceInteractions()
+                .SyncSearchParametersAsync(CancellationToken.None));
+            task.Wait();
```

Comment:
Should we be using: https://learn.microsoft.com/en-us/dotnet/api/system.threading.tasks.task.runsynchronously?view=net-7.0 here?